### PR TITLE
Honor ftp_proxy environment variables in remote_file

### LINKED ISSUE #4216: ftp resource does not respect ENV['ftp_proxy'] setting
It [looks](https://github.com/chef/chef/blob/8d32fdd4377476a9f7dc36a864ccdaa17c32b3a1/lib/chef/provider/remote_file/ftp.rb#L150-L162) like it respects the `Chef::Config` property.  It also [looks](https://github.com/chef/chef/blob/8d32fdd4377476a9f7dc36a864ccdaa17c32b3a1/lib/chef/application.rb#L326-L333) like it exports the `Chef::Config` property to the environment variable.

However, the inverse is not true - it does not natively respect `ENV['ftp_proxy']`

### PR DESCRIPTION
**Pull request by smith**
Fixes #4216.

### CODE REVIEW

File: lib/chef/provider/remote_file/ftp.rb [hunk 1]
```diff
@@ -148,7 +148,7 @@ def get
 
         #adapted from buildr/lib/buildr/core/transports.rb via chef/rest/rest_client.rb
         def proxy_uri(uri)
-          proxy = Chef::Config["ftp_proxy"]
+          proxy = Chef::Config["ftp_proxy"] || ENV["ftp_proxy"] || ENV["FTP_PROXY"]
```
Comment (lamont-granquist) [inline lib/chef/provider/remote_file/ftp.rb]: so why does ftp_proxy not simply default to this?

Comment (lamont-granquist) [inline lib/chef/provider/remote_file/ftp.rb]: yeah, i'd really like to see this added to Chef::Config to pull the config variables out of the environment by default.

needs to be applied to http_proxy and https_proxy and the rest of the proxy universe as well.  it'll fix a whole large class of this kind of bug, and ensure we don't make mistakes like this going forwards.

Comment (smith) [inline lib/chef/provider/remote_file/ftp.rb]: Working on that stuff for a separate upcoming PR.

**Comment by thommay** [comment]
Please feel free to reopen as and when this gets finished :)

**Pull request closed without merging**